srv/src/main/java/my/bookshop/handlers/DestinationConfiguration.java

```diff
@@ -18,15 +22,32 @@
 	@Value("${api-hub.api-key:}")
 	private String apiKey;
 
+	@Autowired
+	private Environment environment;
+
 	@Before(event = ApplicationLifecycleService.EVENT_APPLICATION_PREPARED)
-	public void initializeDestinations() {
+	void initializeDestinations() {
 		if(apiKey != null && !apiKey.isEmpty()) {
 			DefaultHttpDestination httpDestination = DefaultHttpDestination
 					.builder("https://sandbox.api.sap.com/s4hanacloud")
 					.header("APIKey", apiKey)
 					.name("s4-business-partner-api").build();
 
-			DestinationAccessor.prependDestinationLoader(new DefaultDestinationLoader().registerDestination(httpDestination));
+			DestinationAccessor.prependDestinationLoader(
+				new DefaultDestinationLoader().registerDestination(httpDestination));
+		}
+	}
+
+	@EventListener
+	void applicationReady(ApplicationReadyEvent ready) {
+		Integer port = environment.getProperty("local.server.port", Integer.class);
+		if(port != null) {
+			DefaultHttpDestination httpDestination = DefaultHttpDestination
+			.builder("http://localhost:" + port)
+			.name("myself").build();
+
+			DestinationAccessor.prependDestinationLoader(
+				new DefaultDestinationLoader().registerDestination(httpDestination));
 		}
 	}
```

Comment:
For the `cloud` profile, is this guaranteed to happen before `initializeDestinations()`?

Reply:
this most likely happens after `initializeDestinations()` do you see an issue with that?

Reply:
Okay, first I thought they might overwrite each other, but they are actually just added as independent loaders to the DestinationAccessor.

But why did you not just add the new destination loader within the before handler, instead of creating a new event listener for the `ApplicationReadyEvent` event?

Reply:
Or is the `port` property eventually not available yet at that time?

Reply:
> Or is the local.server.port property eventually not available yet at that time?

Exactly :) the Tomcat port is only available very late in the game :)